service-app/internal/aws/client.go

```diff
@@ -2,24 +2,30 @@ package aws
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"io"
+	"strings"
 	"time"
 
 	awsSdk "github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/s3"
 	"github.com/aws/aws-sdk-go-v2/service/s3/types"
 	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
 	"github.com/ministryofjustice/opg-scanning/config"
 )
 
 type AwsClientInterface interface {
 	GetSecretValue(ctx context.Context, secretName string) (string, error)
+	GetSsmValue(ctx context.Context, secretName string) (string, error)
```

Comment:
I don't think `GetSsmValue` is used outside this file, so it could be removed from the interface and made an internal method

Reply:
Its used to mock behaviour in auth package, see auth/middleware_mock.go

`	mockAwsClient.On("GetSsmValue", mock.Anything, "/local/local-credentials").Maybe().Return(fmt.Sprintf("opg_document_and_d@publicguardian.gsi.gov.uk:%s", passwordHash), nil)
`

Reply:
Indeed! Good catch